jsonpatcher/jsonpatcher.go

```diff
@@ -91,7 +91,12 @@ func (jp *jsonPatcher) Create(actions []core.Action) ([]core.Event, format.Node,
 	return events, n, nil
 }
 
-func (jp *jsonPatcher) Reduce(events []core.Event, datastore ds.TxnDatastore, baseKey ds.Key) ([]core.ReduceAction, error) {
+func (jp *jsonPatcher) Reduce(
+	events []core.Event,
+	datastore ds.TxnDatastore,
+	baseKey ds.Key,
+	indexFunc func(model string, key ds.Key, oldData, newData []byte, txn ds.Txn) error,
```

Comment:
As previously discussed, an `indexFun` is used here to index the values within the Event Codec's Reduce function. If we could assume a particular type of data all the time (say, always a document), then we could probably push this indexing step out to the underlying datastore... but since we can't, and want to support alternative Codecs, here is the best place for now...

Reply:
(assuming we want to modify the indexes in the same transaction as the insertions/updates)